pkg/segment/reader/record/recordreader.go

```diff
@@ -433,33 +433,144 @@ func readColsForRecords(segReader *segread.MultiColSegmentReader, blockNum uint1
 	esQuery := false
 	aggs := &structs.QueryAggregators{}
 	nodeRes := &structs.NodeResult{}
-	unorderedResults := readAllRawRecords(orderedRecNums, blockNum, segReader, allMatchedColumns, esQuery, qid, aggs, nodeRes)
-
 	results := make(map[string][]utils.CValueEnclosure)
-	for cname := range allMatchedColumns {
-		results[cname] = make([]utils.CValueEnclosure, 0, len(orderedRecNums))
+
+	for _, colInfo := range segReader.AllColums {
+		cname := colInfo.ColumnName
+
+		if !esQuery && (cname == "_type" || cname == "_id") {
+			continue
+		}
+
+		if cname == config.GetTimeStampKey() {
+			continue
+		}
+
+		results[cname] = make([]utils.CValueEnclosure, len(orderedRecNums))
 	}
 
-	for _, recNum := range orderedRecNums {
-		for cname, value := range unorderedResults[recNum] {
-			if _, ok := results[cname]; !ok {
-				results[cname] = make([]utils.CValueEnclosure, 0)
-			}
-			enclosure := utils.CValueEnclosure{}
-			err := enclosure.ConvertValue(value)
-			if err != nil {
-				log.Errorf("qid=%d, readColsForRecords: failed to convert value %+v for column %v; err=%v",
-					qid, value, cname, err)
-				return nil, err
-			}
+	readAllRawRecordsV2(orderedRecNums, blockNum, segReader,
+		allMatchedColumns, esQuery, qid, aggs, nodeRes, results)
 
-			results[cname] = append(results[cname], enclosure)
+	if aggs != nil && aggs.OutputTransforms != nil {
+		if aggs.OutputTransforms.OutputColumns != nil && aggs.OutputTransforms.OutputColumns.RenameColumns != nil {
+			for oldCname, newCname := range aggs.OutputTransforms.OutputColumns.RenameColumns {
+
+				if results[oldCname] != nil && oldCname != newCname {
+					results[newCname] = results[oldCname]
+					delete(results, oldCname)
+					allMatchedColumns[newCname] = true
+					delete(allMatchedColumns, oldCname)
+				}
+			}
 		}
 	}
 
 	return results, nil
 }
 
+// TODO: remove calls to this function so that only readColsForRecords calls
+// this function. Then remove the parameters that are not needed.
+func readAllRawRecordsV2(orderedRecNums []uint16, blockNum uint16, segReader *segread.MultiColSegmentReader,
```

Comment:
Why did you make a V2 function instead of updating the original function? Was this since the PR is currently Draft and you wanted to easily perf-test V1 vs V2?

Reply:
Renamed older methods to have OldPipeline suffix and newer methods have the default names.